packages/manager/apps/pci-instances/src/data/hooks/instance/selectors/instances.selector.ts

```diff
@@ -0,0 +1,159 @@
+import { InfiniteData } from '@tanstack/react-query';
+import { TInstanceDto, TInstanceStatusDto } from '@/types/instance/api.type';
+import {
+  TAddress,
+  TInstance,
+  TInstanceAction,
+  TInstanceActions,
+  TInstanceAddressType,
+  TInstanceStatus,
+  TInstanceStatusSeverity,
+} from '@/types/instance/entity.type';
+import { TActionName } from '@/types/instance/common.type';
+
+const buildInstanceStatusSeverity = (
+  status: TInstanceStatusDto,
+): TInstanceStatusSeverity => {
+  switch (status) {
+    case 'BUILDING':
+    case 'REBOOT':
+    case 'REBUILD':
+    case 'REVERT_RESIZE':
+    case 'SOFT_DELETED':
+    case 'VERIFY_RESIZE':
+    case 'MIGRATING':
+    case 'RESIZE':
+    case 'BUILD':
+    case 'SHUTOFF':
+    case 'RESCUE':
+    case 'SHELVED':
+    case 'SHELVED_OFFLOADED':
+    case 'RESCUING':
+    case 'UNRESCUING':
+    case 'SNAPSHOTTING':
+    case 'RESUMING':
+    case 'HARD_REBOOT':
+    case 'PASSWORD':
+    case 'PAUSED':
+      return 'warning';
+    case 'DELETED':
+    case 'ERROR':
+    case 'STOPPED':
+    case 'SUSPENDED':
+    case 'UNKNOWN':
+      return 'error';
+    case 'ACTIVE':
+    case 'RESCUED':
+    case 'RESIZED':
+      return 'success';
+    default:
+      return 'info';
+  }
+};
+
+const getInstanceStatus = (status: TInstanceStatusDto): TInstanceStatus => ({
+  state: status,
+  severity: buildInstanceStatusSeverity(status),
+});
```

Comment:
This seems weird to have a method that passes only the status as state, also the 2 properties are correlated